src/clj/rems/api/services/organizations.clj

```diff
@@ -47,14 +51,41 @@
        (find-first (comp #{(:organization/id org)} :organization/id))))
 
 (defn add-organization! [userid org]
-  (organizations/add-organization! userid org))
+  (try
+    (organizations/add-organization! userid org)
+    {:success true
+     :organization/id (:organization/id org)}
+    (catch Exception ex
+      {:success false}
+      (if (and (.getCause ex)
+               (str/includes? (.getMessage (.getCause ex))
+                              "duplicate key value violates unique constraint"))
```

Comment:
I think we used to have a similar check for the resource uniqueness. Could of course first check if the id exists using conventional means using SQL function and leave this exception from the constraint unchecked. It would remove this "nasty code" but it would not test the existence of the constraint. Worthwhile to do?

Reply:
I think you should be able to do something like this on the postgres side:
```
INSERT INTO organizations ....
ON CONFLICT DO NOTHING
RETURNING id
```

Reply:
Yeah, but we actually want to give an error to the end-user here? This would not help in that?

Reply:
you can check the return value, you get `{:id "foo"}` on success and `nil` on conflict, or something like that

Reply:
Sure, I'll try that.

Reply:
Looks nice now!